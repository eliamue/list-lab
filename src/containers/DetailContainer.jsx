import React, { Component } from 'react';
import Game from '../components/games/Game';
import { fetchCreatures } from '../services/ZeldaApi';

export default class DetailContainer extends Component {
  state = {
    loading: true,
    name: '',
    description: '',
    common_locations: [],
    drops: [],
    image: ''
  };

  componentDidMount() {
    fetchCreatures().then(() =>
      this.setState({
        name,
        description,
        image,
        common_locations,
        drops,
        loading: false,
      })
    );
    const { loading, name, description, image, common_locations, drops } =
      this.state;
    if(loading) {
      return (
        <img
          src="https://thumbs.gfycat.com/DistantGrayFinwhale-max-1mb.gif"
          alt="Triforce loading icon"
        />
      );
    } else {
      return (
        <Game
          name={name}
          description={description}
          image={image}
          drops={drops}
          common_locations={common_locations}
        />
      );
    }
  }
}
