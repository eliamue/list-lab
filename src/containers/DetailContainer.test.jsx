import React from 'react';
import { render, screen } from '@testing-library/react';
import DetailContainer from './DetailContainer';
import { MemoryRouter } from 'react-router-dom';

describe('DetailContainer', () => {
  it('renders a single Breath of the Wild creature', async () => {
    render(
      <MemoryRouter initialEntries={['/47']}>
        <DetailContainer
          match={{
            params: {
              id: '47',
            },
          }}
        />
      </MemoryRouter>
    );

    screen.getByAltText('Triforce loading icon');
    const cre = await screen.findByText('Creature', {}, { timeout: 2000 });

    expect(cre).toMatchSnapshot();
  });
});
