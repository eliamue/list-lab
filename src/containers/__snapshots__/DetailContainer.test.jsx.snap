// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailContainer renders a single Breath of the Wild creature 1`] = `
<h2>
  Creature
</h2>
`;
